docs/API-functions.md

```diff
@@ -1614,6 +1618,284 @@ is not availble - it then sets the value to `SRT_REJC_PREDEFINED + 404`.
+
+#### `SRT_ERDVNOSERV`
+
+The required operation cannot be performed when the socket is set
+rendezvous mode (`SRTO_RENDEZVOUS` set to true). Usually it's about
+trying to call `srt_listen` on such socket.
+
+#### `SRT_ERDVUNBOUND`
+
+The socket has been set rendezvous mode (`SRTO_RENDEZVOUS` set to
+true), and was tried to be connected without being bound first. The
+rendezvous connection requires setting up two addresses and ports
+on both sides of the connection, then set local one with `srt_bind`
+and use the remote one with `srt_connect` (or you can simplify it
+using `srt_rendezvous`). Calling `srt_connect*` on an unbound socket
+(in `SRTS_INIT` state), that is, to be bound implicitly, is only allowed
+for regular caller socket (not rendezvous).
+
+#### `SRT_EINVALMSGAPI`
+
+The function was used incorrectly in the message API. This can happen if:
+
+* The parameters specific for the message API in `SRT_MSGCTRL` type parameter
+were incorrectly specified
+
+* The extra parameter check performed by the congestion controller has
+failed
+
+* The socket is a member of a self-managing group, therefore you should
+perform the operation on the group, not on this socket
+
+
+#### `SRT_EINVALBUFFERAPI`
+
+The function was used incorrectly in the stream (buffer) API, that is,
+either the stream-only functions were used with set message API
+(`srt_sendfile`/`srt_recvfile`) or TSBPD mode was used with buffer API
+(`SRTO_TSBPDMODE` set to true) or the congestion controller has failed
+to check call parameters.
+
+#### `SRT_EDUPLISTEN`
+
+The port tried to be bound for listening is already busy. Note that binding
+to the same port is allowed in general (when `SRTO_REUSEADDR` is true on
+every socket that bound it), but only one such socket can be a listener.
+
+#### `SRT_ELARGEMSG`
+
+Size exceeded. This is reported in the following situations:
+
+* Trying to receive a message, but the read-ready message is larger than
+the buffer passed to the receiving function
+
+* Trying to send a message, but the size of this message exceeds the
+size of the preset sender buffer, so it cannot be stored in the sender buffer.
+
+* With getting group data, the array to be filled is too small.
+
+
+#### `SRT_EINVPOLLID`
+
+The epoll ID passed to an epoll function is invalid
+
+#### `SRT_EPOLLEMPTY`
+
+The epoll container has currently no subscribed sockets. This is reported by an
+epoll waiting function that would in this case block forever. This problem
+might be reported both in a situation that you have created a new epoll
+container and didn't subscribe any socket to it, or you did, but all these
+sockets have been closed (including when closed in a separate thread while the
+waiting function was blocking). Note that this situation can be prevented from
+by setting the `SRT_EPOLL_ENABLE_EMPTY` flag, which may be useful in case when
+you use multiple threads and start waiting without subscribed sockets, so that
+you can subscribe them later from another thread.
```

Comment:
```suggestion
The epoll container currently has no subscribed sockets. This is reported by an
epoll waiting function that would in this case block forever. This problem
might be reported both in a situation where you have created a new epoll
container and didn't subscribe any sockets to it, or you did, but these
sockets have been closed (including when closed in a separate thread while the
waiting function was blocking). Note that this situation can be prevented
by setting the `SRT_EPOLL_ENABLE_EMPTY` flag, which may be useful when
you use multiple threads and start waiting without subscribed sockets, so that
you can subscribe them later from another thread.
```